robyn/dependency_injection.py

```diff
@@ -0,0 +1,55 @@
+""" This is Robyn's dependency injection file.
+"""
+from robyn.robyn import (
+    Response,
+    Request,
+)
+
+
+class DependencyMap:
+    def __init__(self):
+        #'request' and 'response' mappings are needed for when constructing deps_to_pass in router.py
+        self._dependency_map = {"ALL": {"request": Request, "response": Response}}
+
+    def add_route_dependency(self, route, **kwargs):
+        """ Adds a dependency to a route.
+        
+        Args:
+            route (str): The route to add the dependency to.
+            kwargs (dict): The dependencies to add to the route.
+        """
+        if route not in self._dependency_map:
+            self._dependency_map[route] = {}
+
+        self._dependency_map[route].update(**kwargs)
+
+        # this is a bit inneficient
+        # we are always adding the global dependencies to the route dependencies
+        # we should be able to to compose the basic data structure in a more efficient way
+        # cc @Darren
```

Comment:
@DarrenSeng

Reply:
I addressed this logic in the most recent commit. Global dependencies no longer get constantly added to route dependencies. The issue I had before you brought this up was that I wasn't sure how handler functions associated with route specific dependencies could still make use of global "ALL_ROUTES" dependencies. In the commit, in router.py, I changed how deps_dict was constructed so that instead of 
`deps_dict = dependencies.get(endpoint, dependencies["ALL_ROUTES"])`
we have
`deps_dict = {**dependencies.get(endpoint, {}), **dependencies["ALL_ROUTES"]}`
This construction combines dependencies specific to the endpoint with the global "ALL" dependencies into a single dictionary deps_dict. If the endpoint is not specified, it will default to an empty dictionary for the endpoint-specific dependencies, effectively only using the global "ALL" dependencies.